components/eventing-controller/pkg/commander/beb/beb.go

```diff
@@ -0,0 +1,191 @@
+	return &Commander{
+		envCfg:          env.GetConfig(),
+		restCfg:         restCfg,
+		enableDebugLogs: enableDebugLogs,
+		metricsAddr:     metricsAddr,
+		resyncPeriod:    resyncPeriod,
+	}
+}
+
+// Init implements the Commander interface.
+func (c *Commander) Init(mgr manager.Manager) error {
+	if len(c.envCfg.Domain) == 0 {
+		return fmt.Errorf("env var DOMAIN must be a non-empty value")
+	}
+	c.mgr = mgr
+	return nil
+}
+
+// Start implements the Commander interface and starts the manager.
+func (c *Commander) Start() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	c.cancel = cancel
+	dynamicClient := dynamic.NewForConfigOrDie(c.restCfg)
+	applicationLister := application.NewLister(ctx, dynamicClient)
+
+	// Need to read env so as to read BEB related secrets
+	c.envCfg = env.GetConfig()
+	reconciler := subscription.NewReconciler(
+		ctx,
+		c.mgr.GetClient(),
+		applicationLister,
+		c.mgr.GetCache(),
+		ctrl.Log.WithName("reconciler").WithName("Subscription"),
+		c.mgr.GetEventRecorderFor("eventing-controller-beb"),
+		c.envCfg,
+	)
+
+	c.backend = reconciler.Backend
+	if err := reconciler.SetupUnmanaged(c.mgr); err != nil {
+		return fmt.Errorf("unable to setup the BEB Subscription Controller: %v", err)
+	}
+	return nil
+}
+
+// Stop implements the Commander interface and stops the commander.
+func (c *Commander) Stop() error {
+	c.cancel()
+
+	return c.cleanup()
+}
+
+// cleanup removes all created BEB artifacts.
+func (c *Commander) cleanup() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	logger := ctrl.Log.WithName("eventing-controller-beb-cleaner").WithName("Subscription")
+	var bebBackend *handlers.Beb
+	var ok bool
+	var bebBackendErr error
+	if bebBackend, ok = c.backend.(*handlers.Beb); !ok {
+		bebBackendErr = errors.New("failed to convert backend to handlers.Beb")
+		logger.Error(bebBackendErr, "no BEB backend exists")
+	}
+
+	// Fetch all subscriptions.
+	dynamicClient := dynamic.NewForConfigOrDie(c.restCfg)
+	subscriptionsUnstructured, err := dynamicClient.Resource(handlers.GroupVersionResource()).Namespace(corev1.NamespaceAll).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	subs, err := handlers.ToSubscriptionList(subscriptionsUnstructured)
+	if err != nil {
+		return err
+	}
+
+	statusDeletionResult := make(map[string]error)
+	subDeletionResult := make(map[string]error)
+	apiRuleDeletionResult := make(map[string]error)
+
+	for _, sub := range subs.Items {
+		// Clean APIRules.
+		apiRule := sub.Status.APIRuleName
+		keyAPIRule := types.NamespacedName{
+			Namespace: sub.Namespace,
+			Name:      apiRule,
+		}
+		if apiRule != "" {
+			err := dynamicClient.Resource(handlers.APIRuleGroupVersionResource()).Namespace(sub.Namespace).Delete(ctx, apiRule, metav1.DeleteOptions{})
+			if err != nil {
+				apiRuleDeletionResult[keyAPIRule.String()] = err
```

Comment:
Why don't we log the error here? Why do we store it so later we can log it?

Reply:
Refactored with in place logging and removed the error maps.